default/AI/ResourcesAI.py

```diff
@@ -29,98 +38,258 @@
-    for pid, planet in planets:
-        newTargets[pid][PFocus] = newTargets[pid][GFocus]
-
-
-def get_resource_target_totals(empirePlanetIDs):
-    pp = sum(x.currentMeterValue(fo.meterType.targetIndustry) for x in planetMap.values())
-    rp = sum(x.currentMeterValue(fo.meterType.targetResearch) for x in planetMap.values())
-    _fill_data_dicts(empirePlanetIDs)
-    return pp, rp
-
-
-def print_resources_priority():
-    """calculate top resource priority"""
-    universe = fo.getUniverse()
-    empire = fo.getEmpire()
-    empirePlanetIDs = PlanetUtilsAI.get_owned_planets_by_empire(universe.planetIDs)
-    print "Resource Management:"
-    print
-    print "Resource Priorities:"
-    resourcePriorities = {}
-    for priorityType in get_priority_resource_types():
-        resourcePriorities[priorityType] = foAI.foAIstate.get_priority(priorityType)
-
-    sortedPriorities = resourcePriorities.items()
-    sortedPriorities.sort(lambda x, y: cmp(x[1], y[1]), reverse=True)
-    topPriority = -1
-    for evaluationPair in sortedPriorities:
-        if topPriority < 0:
-            topPriority = evaluationPair[0]
-        print "    ResourcePriority |Score: %s | %s " % (evaluationPair[0], evaluationPair[1])
-
-    # what is the focus of available resource centers?
-    print
-    warnings = {}
-    print "Planet Resources Foci:"
-    for planetID in empirePlanetIDs:
-        planet = universe.getPlanet(planetID)
-        planetPopulation = planet.currentMeterValue(fo.meterType.population)
-        maxPop = planet.currentMeterValue(fo.meterType.targetPopulation)
-        if maxPop < 1 and planetPopulation > 0:
-            warnings[planet.name] = (planetPopulation, maxPop)
-        statusStr = "  ID: " + str(planetID) + " Name: % 18s -- % 6s % 8s " % (str(planet.name), str(planet.size), str(planet.type))
-        statusStr += " Focus: % 8s" % ("_".join(str(planet.focus).split("_")[1:])[:8]) + " Species: " + str(planet.speciesName) + " Pop: %2d/%2d" % (planetPopulation, maxPop)
-        print statusStr
-    print "\n\nEmpire Totals:\nPopulation: %5d \nProduction: %5d\nResearch: %5d\n" % (empire.population(), empire.productionPoints, empire.resourceProduction(fo.resourceType.research))
-    if warnings != {}:
-        for pname in warnings:
-            mp, cp = warnings[pname]
-            print "Population Warning! -- %s has unsustainable current pop %d -- target %d" % (pname, cp, mp)
+        self.possible_output[self.current_focus] = (itarget, rtarget)
+        self.future_focus = self.current_focus
+
+
+class PlanetFocusManager(object):
+    """PlanetFocusManager tracks all of the empire's planets, what their current and future focus will be."""
+
+    def __init__(self):
+        universe = fo.getUniverse()
+
+        resource_timer.start("getPlanets")
+        planet_ids = list(PlanetUtilsAI.get_owned_planets_by_empire(universe.planetIDs))
+
+        # resource_timer.start("Shuffle")
+        # shuffle(generalPlanetIDs)
+
```

Comment:
It is from the legacy code. 

I think it might have been to test if the algorithm was stable independent of planet id.  It is not.  Changing the order of the pids changes the results.

I'll remove those lines.